pkg/migrate/marketplacemigration.go

```diff
@@ -54,7 +65,32 @@ func (m *MarketplaceMigration) Migrate(ri *v1.ResourceInstance) (*v1.ResourceIns
 		return ri, nil
 	}
 
-	err := m.updateService(ri)
+	ctx := context.WithValue(context.Background(), serviceName, ri.Name)
+
+	// check resource to see if this apiservice has already been run through migration
+	apiSvc, err := ri.AsInstance()
+	if err != nil {
+		return nil, err
+	}
+
+	// get x-agent-details and determine if we need to process this apiservice for marketplace provisioning
+	details := util.GetAgentDetails(apiSvc)
+	if len(details) > 0 {
+		completed := details[definitions.MarketplaceMigration]
+		if completed == definitions.MigrationCompleted {
+			// migration ran already
+			m.logger.
+				WithField(string(serviceName), apiSvc).
+				Debugf("marketplace provision migration already completed")
```

Comment:
Can we please remove the log so we don't see this each time we start the agent, and just log when we do migrate a service